modules/core/src/main/scala/com.snowplowanalytics.snowplow.bigquery/processing/Processing.scala

```diff
@@ -179,28 +182,32 @@ object Processing {
     loadTstamp: java.lang.Long,
     events: ListOfList[Event],
     v2Entities: NonAtomicFields.Result,
-    legacyEntities: LegacyColumns.Result
+    legacyEntities: LegacyColumns.Result,
+    legacyColumnMode: Boolean
   ): F[(List[BadRow], List[EventWithTransform])] =
     Foldable[ListOfList]
       .traverseSeparateUnordered(events) { event =>
         Sync[F].delay {
-          val v2Transform = Transform
-            .transformEvent[AnyRef](badProcessor, BigQueryCaster, event, v2Entities)
-            .map { namedValues =>
-              namedValues
-                .map { case Caster.NamedValue(k, v) =>
-                  k -> v
-                }
-                .toMap
-                .updated("load_tstamp", loadTstamp)
-            }
-          val legacyTransform = LegacyColumns
-            .transformEvent(badProcessor, event, legacyEntities)
-
-          for {
-            map1 <- v2Transform
-            map2 <- legacyTransform
-          } yield event -> (map1 ++ map2)
+          if (legacyColumnMode) LegacyColumns.transformEvent(badProcessor, event, legacyEntities).map(event -> _)
```

Comment:
This line doesn't look right.  I might be wrong... but are you sure you're loading the atomic fields, like `event_id` and `collector_tstamp` etc?

Reply:
I'll add a test for that